extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -0,0 +1,437 @@
+					args.firstUser = notification.latestActors[0].name;
+					args.secondUser = notification.latestActors[1].name;
+				} else {
+					args.user = notification.latestActors[0].name;
+				}
+
+				return fillArgs(message, args);
+			};
+
+			this._getReplyKey = function (title, totalUniqueActors) {
+				const user = totalUniqueActors <= 1 ? '' :
+					totalUniqueActors == 2 ? 'two-users-' : 'multiple-users-';
+				return 'notifications-replied-by-' + user + (title ? 'with-title' : 'no-title');
+			};
+
+			this._getPostUpvoteText = function (notification) {
+				const key = 'notifications-post-upvote' + this._getUpvoteKey(notification.title, notification.totalUniqueActors);
+				const message = window.mw.message(key).parse();
+				return fillArgs(message, {postTitle: bold(notification.title)});
+			};
+
+			this._getReplyUpvoteText = function (notification) {
+				const key = 'notifications-reply-upvote' + this._getUpvoteKey(notification.title, notification.totalUniqueActors);
+				const message = window.mw.message(key).parse();
+				return fillArgs(message, {postTitle: bold(notification.title)});
+			};
+
+			this._getUpvoteKey = function (title, totalUniqueActors) {
+				return '-' + (totalUniqueActors <= 1 ? 'single-user' : 'multiple-users')
+					+ '-' + (title ? 'with-title' : 'no-title');
+			};
+		}
+
+		function View(logic, template, textFormatter) {
+			this.logic = logic;
+			this.template = template;
+			this.textFormatter = textFormatter;
+			this.$notificationsCount = $('#on-site-notifications-count');
+			this.$container = $('#notification-container');
+			this.$markAllAsReadButton = $('#mark-all-as-read-button');
+
+			this.registerEvents = function () {
+				this.addDropdownLoadingEvent();
+				this.addMarkAllAsReadEvent();
+			};
+
+			this.addMarkAllAsReadEvent = function () {
+				this.$markAllAsReadButton.click($.proxy(this.logic.markAllAsRead, this.logic));
+			};
+
+			this.addDropdownLoadingEvent = function () {
+				var $dropdown = $('#on-site-notifications-dropdown');
+				$dropdown.click(function () {
+					logic.loadFirstPage();
+				});
+			};
+
+			this._mapToView = function (notifications) {
+				function getIcon(type) {
+					if (type === notificationTypes.discussionReply) {
+						return 'wds-icons-reply-small';
+					} else if (type === notificationTypes.announcement) {
+						return 'wds-icons-megaphone';
+					} else {
+						return 'wds-icons-upvote-small';
+					}
+				}
+
+				function getAvatars(actors) {
+					return actors.slice(0, 5);
+				}
+
+				return notifications.map(this.proxy(function (notification) {
+					return {
+						icon: getIcon(notification.type),
+						uri: notification.uri,
+						snippet: notification.snippet,
+						text: this.textFormatter.getText(notification),
+						isUnread: notification.isUnread,
+						communityName: notification.communityName,
+						showAvatars: notification.totalUniqueActors > 2 && notification.type === notificationTypes.discussionReply,
```

Comment:
For future please consider returning this data by backend.
As we discussed offline - please make sure it's ok that we return list of upvoters in json response that is not visible.